packages/core/src/html/linkProcessor.js

```diff
@@ -87,18 +87,34 @@ function convertMdAndMbdExtToHtmlExt(node) {
     if (hasNoConvert) {
       return;
     }
+
     const { href } = node.attribs;
-    const hasMdExtension = href.slice(-3) === '.md';
-    if (hasMdExtension) {
-      const newHref = `${href.substring(0, href.length - 3)}.html`;
-      node.attribs.href = newHref;
+
+    const hasMdExtension = href.includes('.md');
```

Comment:
let's enhance the check since we're supporting hashes now: e.g. you may have a false positive `x.md.jpg`

Reply:
Would this work better? 

```
    const hasMdExtension = href.slice(-3) === '.md' || href.includes('.md#');
    const hasMbdExtension = href.slice(-4) === '.mbd' || href.includes('.mbd#');
```

The above should handle the case that you suggested, `x.md.jpg`. 

Typically, only one of the two conditions will be true (XOR). 

The edge case is if the link has both `md` and `mbd` extension, which can fall into the following cases:
1) .md#.mbd
2) .mbd#.md
3) both .md# and .mbd# appears somewhere in the link

I think cases 1 and 2 are still manageable but should we handle case 3?

Reply:
hmm would just the old check with https://github.com/MarkBind/markbind/pull/1493#discussion_r588378063 work for all 3 cases?

Reply:
Yup, using hash function in the URL library would work. Thanks for the advice!